### ISSUE TITLE
Activity opens and closes immediately

### ISSUE THREAD

**Issue by Dimi20cen** [163gal/Lemonade#12]
Activity seems to open to grey screen and closes a second after openning. It doesn't log any errors. Tested on sugar live-build. With `python3 LemonadeStand.py` however it runs.

@sourabhaa @chimosky @jriyyya, Could you please check if it's reproducable in your environments?

**Comment by jriyyya** [comment]
It is working fine with my environment, with the same sugar live-build. Started the activity using `sugar-activity3 .`

**Comment by quozl** [comment]
https://github.com/sugarlabs/Lemonade/blob/d198c35e1013a5fa1015418d8c6bfafcc6f77053/activity.py#L73 requires that `python` be in path, which isn't always the case.  Some systems require `python3` instead.

The activity should be ported away from the Pippy wrapper it was built within, and instead use _sugargame_.  As a result, it should start faster because a subprocess and second application window are no longer required.

**Comment by Dimi20cen** [comment]
> Rrequires that python be in path

Thank you, that resolved the issue.

> The activity should be ported away from the Pippy wrapper it was built within, and instead use sugargame

I'm currently in the process of porting the activity to Sugargame and have successfully gotten it to run as intended, at least on a surface level. Regarding the console functionality that was previously present, should it be completely removed or reimplemented in a different manner? Apologies if this question seems dumb.

**Comment by chimosky** [comment]
> I'm currently in the process of porting the activity to Sugargame and have successfully gotten it to run as intended, at least on a surface level. Regarding the console functionality that was previously present, should it be completely removed or reimplemented in a different manner? Apologies if this question seems dumb.

Porting shouldn't remove existing functionality, if this functionality was dependent on the component being ported from, then you can look at implementing said functionality for the activity and making it work with sugargame.

**Comment by Dimi20cen** [comment]
Since the `DEBUG_TERMINAL` is already set to False, porting the activity to pygame and not implementing another type of debugging console will not change the user experience.

**Comment by Dimi20cen** [comment]
Or I have a different idea, what if when `DEBUG_TERMINAL=False` the activity is just as before with the pippy wrapper. But when `DEBUG_TERMINAL=True`, the activity is pygame wrapped.

**Comment by chimosky** [comment]
> Since the DEBUG_TERMINAL is already set to False, porting the activity to pygame and not implementing another type of debugging console will not change the user experience.

I don't think the debug feature was meant for users but I haven't taken a look so I'm not really sure.


> Or I have a different idea, what if when `DEBUG_TERMINAL=False` the activity is just as before with the pippy wrapper. But when `DEBUG_TERMINAL=True`, the activity is pygame wrapped.

What would the benefits of that be?

**Comment by Dimi20cen** [comment]
Well, if I understand correctly, as quozl said the activity should be ported from pippy to pygame. Now, instead of using the DEBUG_TERMINAL, the log activity could be used as well, just like in other activities.

If however there is the need of keeping the function of DEBUG_TERMINAL alive, then a hybrid approach might be better. 
- DEBUG_TERMINAL=True, debug terminal's function is left untouched
- DEBUG_TERMINAL=False, the activity is wrapped in pygame instead as quozl said

**Comment by quozl** [comment]
I don't think it matters whether DEBUG_TERMINAL is kept or not.  As the developer, you get to decide.  I'll wait to see your commits before I give an opinion.

**Comment by chimosky** [comment]
@Dimi20cen as you have this opened, it'll be a great idea to fix this first before any other PRs you have open.

**Comment by Dimi20cen** [comment]
> I'll wait to see your commits before I give an opinion
> @Dimi20cen as you have this opened, it'll be a great idea to fix this first before any other PRs you have open.

Just opened a draft PR, I would appreciate it if you gave it a look, thank you!

Note: I maintained the console funtion and removed the DEBUG_TERMINAL
